lib/apiconfig/load.go

```diff
@@ -23,28 +27,37 @@ func LoadClientConfig(filename string) (*CalicoAPIConfig, error) {
 			return nil, err
 		}
 
-		c, err := LoadClientConfigFromBytes(b)
+		c, err = LoadClientConfigFromBytes(b)
 		if err != nil {
 			return nil, fmt.Errorf("syntax error in %s: %v", filename, err)
 		}
-		return c, nil
+
+	} else {
+		c,err = LoadClientConfigFromEnvironment()
+	}
+
+	// If EtcdEndpoints is set and DatastoreType is missing set it to EtcdV3
+	// otherwise set default Datastoretype to Kubernetes
+	if c.Spec.DatastoreType == "" && c.Spec.EtcdEndpoints != "" {
+		c.Spec.DatastoreType = EtcdV3
+	} else if c.Spec.DatastoreType == "" {
+		c.Spec.DatastoreType = Kubernetes
 	}
-	return LoadClientConfigFromEnvironment()
+	// Changing Kubeconfig prevents a default error attitude
+	// when user has set no configuration.
+	if c.Spec.DatastoreType == Kubernetes && c.Spec.Kubeconfig == "" {
```

Comment:
Using this logic makes breaks running calicoctl inside a pod. `thanks to @caseydavenport for point it out`

Reply:
hehe, sorry I didn't get around to actually posting a comment on here!

I think we also need to check for the presence of the in cluster config env vars, and only set kubeconfig if those aren't set. WDYT?

Reply:
No worries, Alex told me about it and I got excited :D

Are we talking about `CALICO_CTL_CONTAINER` or something like `KUBERNETES_PORT`?

We could also check for secrets folder. WDYT?
`/var/run/secrets`

Reply:
I was thinking KUBERNETES_SERVICE_HOST / KUBERNETES_SERVICE_PORT